home/service/search.py

```diff
@@ -76,6 +76,42 @@ def _get_context(self) -> dict[str, Any]:
             "paginator": self.paginator,
             "total_results": self.results.total_results,
             "label_clear_href": label_clear_href,
+            "readable_match_reasons": self._get_match_reason_display_names(),
         }
 
         return context
+
+    def _highlight_results(self):
+        "Take a SearchResponse and add bold markdown where the query appears"
+        string_to_highlight = self.form.cleaned_data.get("query") if self.form.is_valid() else ""
+        highlighted_results = deepcopy(self.results)
+
+        if string_to_highlight == "":
+            return highlighted_results
+
+        else:
+            for result in highlighted_results.page_results:
+                metadata = getattr(result, "metadata")
+                metadata["search_summary"] = metadata.get("search_summary", "").replace(
+                    string_to_highlight, f"**{string_to_highlight}**"
+                )
+                setattr(result, "metadata", metadata)
+
+                name = getattr(result, "description")
+                highlighted_description = name.replace(
```

Comment:
There might be conflicts here between markup added by whoever wrote the descriptions and the markup we're adding for the match highlighting.

To avoid this we could
- make sure we strip existing markdown from the description before we repurpose it for the search summaries (this is the case for the ones we've done manually
- make sure `string_to_highlight` doesn't contain any markdown significant characters (such as dashes) before using it here